### ISSUE TITLE
Authorize new user

### ISSUE THREAD

**Issue by traponto** [David-Desmaisons/DiscogsClient#4]
I'm having trouble activating the authorize procedure inside my WinForm application, it seems that the 'GetToken' function is not invoked and the program goes in loop in the 'Authorize' procedure. I took the same instructions in the sample application 'DiscogsAuthenticationConsole' where instead everything is fine.

**Comment by David-Desmaisons** [comment]
Well, It is very hard to diagnostic anything from what you describe. I would say since `DiscogsAuthenticationConsole` is working try to simplify your code to make it more and more similar to the sample application till it works.

**Comment by FrancisChung** [comment]
Sounds like you need to synchronize your UI thread context.

https://stackoverflow.com/questions/19028374/accessing-ui-controls-in-task-run-with-async-await-on-winforms

**Comment by David-Desmaisons** [comment]
@FrancisChung  DiscogsClent is (and needs to be) agnostic in term of application context (WPF, or WinForm or console, or ASP etc...). So it is the caller responsability to make sure that the API is called on the "context" context.

**Comment by traponto** [comment]
Thank you for your answers and for the suggestions you gave me. I took a look at the stackoverflow link (as FrancisChung suggested) and I think I fall into the "deadlock" case described there by "Daniel". I tried to insert the 'Authorize' procedure in a separate thread and I verified that it works. However, I would like to be able to manage the Key (provided by DiscoGS) and tokens (provided by DiscoGSClient) within my WinForm application, so any suggestions to do this are welcome.
